core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -271,31 +272,22 @@ class TransactionStateManager(brokerId: Int,
    * When this broker becomes a follower for a transaction log partition, clear out the cache for corresponding transactional ids
    * that belong to that partition.
    */
-  def removeTransactionsForPartition(partition: Int) {
+  def removeTransactionsForTxnTopicPartition(partitionId: Int) {
     validateTransactionTopicPartitionCountIsStable()
 
-    val topicPartition = new TopicPartition(Topic.TransactionStateTopicName, partition)
-
-    inLock(stateLock) {
-      ownedPartitions.remove(partition)
-      loadingPartitions.remove(partition)
-    }
+    val topicPartition = new TopicPartition(Topic.TransactionStateTopicName, partitionId)
```

Comment:
In step 2 isn't the log append callback happening on another thread? So how do we release the read lock? I guess i'm missing something but i'm reading it like this:
```
inReadLock(lock) {
   // do some stuff
   txnManager.appendToLog(...)
}
```
Which would mean the read lock would be given up as soon as the `appendToLog` call returns, but that doesn't mean the write to the log has actually completed as it might go into purgatory and be completed by another thread. I guess you have some other idea?